AirCasting/SearchAndFollow/CompleteScreen/Views/CompleteScreenViewModel.swift

```diff
@@ -44,41 +81,42 @@ class CompleteScreenViewModel: ObservableObject {
     let sessionEndTime: Date
     let sensorType: String
     @Published var sessionStreams: Loadable<[SessionStreamViewModel]> = .loading
-    @Published var chartViewModel = SearchAndFollowChartViewModel(stream: nil)
+    @Published var chartViewModel = SearchAndFollowChartViewModel()
     
-    private let session: SearchSessionResult
-    @Injected private var service: SearchSessionStreamsDownstream
+    private let session: PartialExternalSession
+    private var service = DefaultStreamDownloader()
+    private var isPresented: Binding<Bool>
+    @Injected private var thresholdsStore: ThresholdsStore
+    @Injected private var singleSessionDownloader: SingleSessionDownloader
     
-    init(session: SearchSessionResult) {
+    init(session: PartialExternalSession, isPresented: Binding<Bool>) {
         self.session = session
         sessionLongitude = session.longitude
         sessionLatitude = session.latitude
         sessionName = session.name
         sessionStartTime = session.startTime
         sessionEndTime = session.endTime
-        sensorType = "OpenAir"
+        sensorType = session.provider
+        self.isPresented = isPresented
         reloadData()
     }
     
     private func reloadData() {
         sessionStreams = .loading
-        service.downloadSession(uuid: session.uuid) { [weak self] result in
+        thresholdsStore.getThresholdsValues(for: Self.getSensorName(session.sensorName)) { [weak self] result in
             guard let self = self else { return }
             switch result {
+            case .success(let thresholdsValues):
+                self.getMeasurementsAndDisplayData(thresholdsValues)
             case .failure(let error):
-                Log.error("Failed to download session: \(error)")
-                self.alert = InAppAlerts.failedSessionDownloadAlert()
-            case .success(let downloadedSession):
-                self.sessionStreams = .ready(
-                    downloadedSession.streams.map({
-                        .init(id: $0.id,
-                              sensorName: Self.showStreamName($0.sensorName),
-                              lastMeasurementValue: $0.measurements.last?.value ?? 0)
-                    })
-                )
-                self.selectedStream = downloadedSession.streams.first?.id
-                if let stream = downloadedSession.streams.first {
-                    self.chartViewModel.setStream(to: stream)
+                switch error {
+                case .noThresholdsFound:
```

Comment:
hmm, makes me think if this "no thresohlds" scenario is really an error. Maybe we should change the store to be able to successfully return nil value? Smth like `Result<Thresholds?, Error> `